schema/query/projection_evaluator.go

```diff
@@ -115,14 +115,14 @@ func evalProjection(ctx context.Context, p Projection, payload map[string]interf
 				if err != nil {
 					return nil, err
 				}
-				rbr.request(ref.Path, q, func(payloads []map[string]interface{}, validator schema.Validator) (err error) {
+				rbr.request(ref.Path, q, &p[pidx], func(payloads []map[string]interface{}, validator schema.Validator, pf *ProjectionField) (err error) {
 					for i := range payloads {
 						if payloads[i], err = evalProjection(ctx, pf.Children, payloads[i], validator, rbr); err != nil {
 							return fmt.Errorf("%s: error applying projection on sub-resource item #%d: %v", pf.Name, i, err)
 						}
 					}
 					var v interface{}
-					if v, err = resolveFieldHandler(ctx, pf, def, payloads); err != nil {
```

Comment:
If I understand the data race correctly, you don't need to do all that to workaround it. You just need to copy the Projection at the beginning of the for loop:

Instead of:

```go
for _, pf := range p {
```

do:

```go
for i := range p {
    pf = p[i]
```

or less readable:

```go
for _, pf := range p {
    pf := pf
```

On a side note, the complexity of this function is thru the roof. We might want to break it down at some point.